src/quo2/components/community/token_gating.cljs

```diff
@@ -1,295 +1,45 @@
-   :token-row                      {:flex-direction :row
-                                    :flex-wrap      :wrap
-                                    :align-items    :center}
-   :token-row-container            {:flex-direction    :row
-                                    :align-items       :center
-                                    :margin-horizontal (- token-tag-horizontal-spacing)}
-   :token-row-container-no-shift   {:margin-left 0}
-   :token-row-or-text              {:margin-right 3}
-   :token-tag-spacing              {:margin-horizontal token-tag-horizontal-spacing
-                                    :margin-vertical   token-tag-vertical-spacing}
-   :divider                        {:height            1
-                                    :margin-horizontal -20
-                                    :margin-top        (- 12 token-tag-vertical-spacing)
-                                    :margin-bottom     8}
-   :membership-request-denied      {:margin-top 13}
-   :enter-button                   {:margin-top 16}
-   :info-text                      {:margin-top         12
-                                    :text-align         :center
-                                    :padding-horizontal 20}})
-
-(defn multiple-token-requirements?
-  [token-requirement-lists]
-  (when (vector? (first token-requirement-lists))
-    (> (count token-requirement-lists) 1)))
-
-(defn is-token-requirement-met?
-  [token-requirement-list]
-  (and (seq token-requirement-list)
-       (every?
-        (fn [token]
-          (get token :is-sufficient?))
-        token-requirement-list)))
-
-(defn are-multiple-token-requirements-met?
-  [token-requirement-lists]
-  (if (multiple-token-requirements? token-requirement-lists)
-    (some is-token-requirement-met? token-requirement-lists)
-    (is-token-requirement-met? token-requirement-lists)))
+            [clojure.string :as string]
+            [quo2.components.community.style :as style]))
 
 (defn token-requirement-list-row
-  [tokens community-color]
+  [tokens padding?]
   [rn/view
-   {:style (merge
-            (get styles :token-row)
-            (get styles :token-requirement-list-spacing))}
-   (map-indexed (fn [token-index token]
-                  (let [{:keys [token-img-src token amount is-sufficient? is-purchasable?]} token]
-                    ^{:key token-index}
-                    [rn/view {:style (get styles :token-tag-spacing)}
-                     [token-tag/token-tag
-                      {:token          token
-                       :value          amount
-                       :size           24
-                       :border-color   community-color
-                       :is-required    is-sufficient?
-                       :is-purchasable is-purchasable?
-                       :token-img-src  token-img-src}]]))
-                tokens)])
+   {:style (style/token-row padding?)}
+   (doall
```

Comment:
do you need to `doall` here? I thought it would be called

Reply:
it always better to have doall in hiccup